File: cpp/n28_strStr.cpp
// https://leetcode-cn.com/problems/implement-strstr/
#include <iostream>
#include <string>
using namespace std;

class Solution {
public:
    int strStr(string haystack, string needle) {
        return haystack.find(needle);
    }
};

int main(){
    Solution solu;
    string haystack = "hello";
    string needle = "ll";
    cout << solu.strStr(haystack, needle) << endl;
}

File: python/n28_strStr.py
# https://leetcode-cn.com/problems/implement-strstr/
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if needle == "":
            return 0
        if needle not in haystack:
            return -1
        return haystack.index(needle)


if __name__ == "__main__":
    solu = Solution()
    haystack = "hello"
    needle = "ll"
    print(solu.strStr(haystack, needle))